static/js/services/search.js

```diff
@@ -102,6 +102,11 @@ var _ = require('underscore'),
 
         return _search(type, filters, options, extensions)
           .then(function(searchResults) {
+            if (docIds && docIds.length) {
+              docIds.forEach(function(docId) {
+                return searchResults.indexOf(docId) === -1 && searchResults.push(docId);
```

Comment:
I think this would be clearer if it was a standard `if` block rather than a shortcircuiting return.